spec/prometheus/client/push_spec.rb

```diff
@@ -111,12 +142,99 @@
       expect(http).to receive(:use_ssl=).with(false)
       expect(http).to receive(:open_timeout=).with(5)
       expect(http).to receive(:read_timeout=).with(30)
-      expect(http).to receive(:request).with(request)
+      expect(http).to receive(:request).with(request).and_return(response)
       expect(Net::HTTP).to receive(:new).with('localhost', 9091).and_return(http)
 
       push.send(:request, Net::HTTP::Post, registry)
     end
 
+    context 'for a 3xx response' do
```

Comment:
I was on the fence about making shared examples for these. On the one hand there's 3 very similar tests that can be parameterised. On the other, there are exactly 3 cases and that won't grow.

Overall I'm not a huge fan of these tests. One thought I had to reduce the level of repetition is to pull `request` and `http` into `let` blocks with all the appropriate method stubs set up. After that, each test could set the up the method call expectations that it specifically cares about (e.g. you'd only set up the `use_ssl` expectation in the test for SSL).

If you think that'd be better I'm happy to follow up either with another commit on this branch, or with a separate refactor PR.

Reply:
I think this is fine?
i'm not a big fan of "shared examples", they tend to make things harder to both read and modify.
Agree you can extract those two very mocked objects to centralize them, if you feel strongly about it. Otherwise, I'd leave this as is until one of those blocks needs to be modified 3 times, and then i'd extract.

Reply:
Yeah, shared examples specifically I'm not a huge fan of either. I think I'm going to go the route of extracting as much of the duplication as possible and not setting up the same mock expectations in every test.

I'll do that in a separate PR though, so we can get this one merged.